ZelBack/src/services/systemService.js

```diff
@@ -567,6 +567,55 @@ async function mongoDBConfig() {
   }
 }
 
+// eslint-disable-next-line consistent-return
+async function mongodGpgKeyVeryfity() {
+  log.info('MongoDB GPG verification...');
+  try {
+    const { stdout, stderr, error } = await serviceHelper.runCommand('gpg', { runAsRoot: true, params: ['--show-keys', '/usr/share/keyrings/mongodb-archive-keyring.gpg'] });
+    if (error) {
+      throw new Error(`Executing gpg: ${error}`);
+    }
+    if (stderr) {
+      throw new Error(`gpg stderr: ${stderr}`);
+    }
+    const expiredMatch = stdout.match(/\[expired: (\d{4}-\d{2}-\d{2})\]/);
+    const versionMatch = stdout.match(/MongoDB (\d+\.\d+) Release Signing Key/);
+    if (expiredMatch) {
```

Comment:
this is only executed if the key is expired right. So we are not doing the network requests actually.
Are we ok on running this function on every restart?

Reply:
The local key is checked on every restart, and the key is only updated if it has expired.